### ISSUE TITLE
Unable to determine Dialect for Oracle 18c XE

### ISSUE THREAD

**Issue by pmverma** [vladmihalcea/hypersistence-optimizer#57]
```
2019-11-05 23:09:19.959  WARN 7304 --- [           main] Hypersistence Optimizer                  : Could not resolve the suggested Dialect. Some database checks will be disabled.

org.hibernate.HibernateException: Unable to determine Dialect to use [name=Oracle, majorVersion=18]; user must register resolver or explicitly set 'hibernate.dialect'
	at org.hibernate.engine.jdbc.dialect.internal.DialectFactoryImpl.determineDialect(DialectFactoryImpl.java:109)
	at org.hibernate.engine.jdbc.dialect.internal.DialectFactoryImpl.buildDialect(DialectFactoryImpl.java:54)
	at io.hypersistence.optimizer.hibernate.scanner.Hibernate53ConfigurationScanner.getSuggestedDialect(Hibernate53ConfigurationScanner.java:173)
	at io.hypersistence.optimizer.hibernate.scanner.Hibernate53ConfigurationScanner.access$100(Hibernate53ConfigurationScanner.java:83)
	at io.hypersistence.optimizer.hibernate.scanner.Hibernate53ConfigurationScanner$2.execute(Hibernate53ConfigurationScanner.java:135)
	at io.hypersistence.optimizer.core.action.ActionExecutor.execute(ActionExecutor.java:70)
	at io.hypersistence.optimizer.hibernate.scanner.Hibernate53ConfigurationScanner.scanDialect(Hibernate53ConfigurationScanner.java:132)
	at io.hypersistence.optimizer.hibernate.scanner.Hibernate53ConfigurationScanner.scan(Hibernate53ConfigurationScanner.java:105)
	at io.hypersistence.optimizer.HypersistenceOptimizer.init(HypersistenceOptimizer.java:74)
	at com.mycompany.myapp.config.HypersistenceConfiguration.run(HypersistenceConfiguration.java:26)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:781)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:765)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:319)
	at com.mycompany.myapp.JhipsterApp.main(JhipsterApp.java:63)

```

**Comment by vladmihalcea** [comment]
Thanks for creating the issue. Can you provide me a test case replicating it. Is it a JHipster application you could share with me to debug the issue? Thanks.

**Comment by pmverma** [comment]
This is the repo I am using, https://github.com/pmverma/jhipster-hypersistence-oracle.
`prod` profile has been activated by default.
Configurations are inside `application-prod.yml`
Only need to run `./mvnw`

**Comment by vladmihalcea** [comment]
Thanks. I'll check it out when I have some time.

**Comment by vladmihalcea** [comment]
@pmverma This issue is caused by the [HHH-13184](https://hibernate.atlassian.net/browse/HHH-13184) Hibernate bug. It was fixed in 5.4, but it's up to @gbadner to integrate it in 5.3. I see there's [this PR](https://github.com/hibernate/hibernate-orm/pull/3018) for backporting it to 5.3.

Anyway, this does not prevent Hypersistence Optimizer from scanning your configurations. It's just that, since the `Dialect` cannot be resolved, it cannot determine whether it matches the one the database server is using.

So, I'm going to close this issue and let's hope the HHH-13184 issue will be backported to the 5.3 branch as well.
